src/app.js

```
@@ -1 +1,42 @@
+/* eslint-disable no-console */
 'use strict';
+
+const fs = require('fs/promises');
+
+async function copy(oldPath, newPath) {
+  if (!oldPath || !newPath) {
+    console.error('Source and destination file paths are required');
+  }
+
+  if (oldPath === newPath) {
+    return;
+  }
+
+  try {
+    const srcStats = await fs.stat(oldPath);
+
+    if (!srcStats.isFile()) {
+      console.error('Source is not a regular file');
+    }
+
+    await fs.copyFile(oldPath, newPath);
+  } catch (err) {
+    console.error('Copy error:', err.message);
```

The catch block logs the error but swallows it. After `console.error('Copy error:', err.message)` you should rethrow the error (or `throw err`) so the caller/test can observe the failure and the process can exit appropriately. Swallowing errors will hide test failures (Checklist 3.5) .